pkg/provider/provider.go

```diff
@@ -6,13 +6,17 @@ import (
 	"net"
 	"strconv"
 
+	"github.com/containous/maesh/pkg/annotations"
 	"github.com/containous/maesh/pkg/k8s"
 	"github.com/containous/maesh/pkg/topology"
 	"github.com/containous/traefik/v2/pkg/config/dynamic"
 	"github.com/sirupsen/logrus"
 	corev1 "k8s.io/api/core/v1"
 )
 
+// MiddlewareBuilder is capable of building a middleware from service annotations.
+type MiddlewareBuilder func(annotations map[string]string) (*dynamic.Middleware, error)
```

Comment:
Why not keep the previous type signature? 

```suggestion
type MiddlewareBuilder func(*topology.Service) (*dynamic.Middleware, error)
```

Reply:
It removes the dependency between topology and annotations. I tried to find an example of a middleware which wouldn't be based on annotations and I couldn't find one.